assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -99,7 +102,41 @@
     private ServerConfigurationService serverConfigurationService;
     private UserDirectoryService userDirectoryService;
     private UserTimeService userTimeService;
+    private FormattedText formattedText;
 
+    //Builds TimeSheet error/success messages
+    @Setter
+    @Getter
+    public class BuildTimeSheetReturnMessage implements Serializable {
```

Comment:
This message pattern seems somewhat complex for something so simple

Reply:
Adrian requested to change the way return messages were created since our method would lead to failures.

As an example, lines 611 .. 617:
```
User u;
		try {
			u = userDirectoryService.getUser(userId);
		} catch (UserNotDefinedException e2) {
            log.warn("You need to be logged in to add time sheet register");
            return "{\"error\": {"
            		+ "\"code\": 1,"
            		+ "\"message\": \"ts.add.err.userId\""
            		+ "}"
            		+ "}";
		}
```
would end up being like that:

```
        User u;
        try {
            u = userDirectoryService.getUser(userId);
        } catch (UserNotDefinedException e2) {
            log.warn("You need to be logged in to add time sheet register");
            return new BuildTimeSheetReturnMessage(false, 1, "ts.rem.err.userId");
        }
```